[ADD] Payroll Contribution Register Report

saltonmassally opened the pull request:
This addon solves two things about printing contribution registers that is undesirable:
- You can only print contribution registers via configuration > contribution
- If you have two runs that overlap in date it is impossible to print run specific contributionn register
  Consequently this addon allowa you to print contibution registers for payslip lines in the run directly from the payslip

saltonmassally committed:
[ADD] Payroll Contribution Register Report
This addon solves two things about printing contribution registers that is undesirable:
* You can only print contribution registers via configuration > contribution
* If you have two runs that overlap in date it is impossible to print run specific contributionn register
Consequently this addon allowa you to print contibution registers for payslip lines in the run directly from the payslip

saltonmassally committed:
fix pep8

dufresnedavid commented on hr_payroll_report_contrib_register/report/__init__.py:
```diff
@@ -0,0 +1,22 @@
+###############################################################################
+import report_contribution_register
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
don't put your vim commands please.

dufresnedavid commented on hr_payroll_report_contrib_register/report/report_contribution_register.py:
```diff
@@ -0,0 +1,91 @@
+    _template = 'hr_payroll.report_contributionregister'
+    _wrapped_report_class = contribution_register_report
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
same here

dufresnedavid commented on hr_payroll_report_contrib_register/report/report_contribution_register.py:
```diff
@@ -0,0 +1,91 @@
+        res = []
+        self.regi_total = 0.0
+        self.cr.execute("SELECT pl.id from hr_payslip_line as pl "
+                        "LEFT JOIN hr_payslip AS hp on (pl.slip_id = hp.id) "
```
instead of using LEFT JOIN, i would use a simple WHERE statement. This way, you let postgres decide how to make the join.

saltonmassally replied:
I copied this from the core hr_payroll module added the where clause to restrict by payslip ids... SQL is not my strength, I tried your method but the payslip lines returned are being duplicated... please supply a snippet if you can

dufresnedavid replied:
SELECT UNIQUE ...

dufresnedavid commented on hr_payroll_report_contrib_register/report/report_contribution_register.py:
```diff
@@ -0,0 +1,91 @@
+
+class wrapped_report_contribution_register(osv.AbstractModel):
+    _name = \
+        'report.hr_payroll_report_contrib_register.report_contributionregister'
```
personnaly, I prefer parenthesis over the backslash

saltonmassally committed:
Cosmetic improvements
* remove vim commands
* remove blacklash in favor for parenthesis

saltonmassally committed:
[IMP] hr_payroll_contrib_register camel casing

saltonmassally committed:
[IMP] replace relative import

feketemihai commented on hr_payroll_report_contrib_register/README.rst:
```diff
@@ -0,0 +1,52 @@
+
+Usage
+=====
+
```
Please add the Try me on Runbot section.

feketemihai commented on hr_payroll_report_contrib_register/README.rst:
```diff
@@ -0,0 +1,52 @@
+Known issues / Roadmap
+======================
+
+None
```
Please add the bug tracker paragraph with predefined template.

feketemihai commented on hr_payroll_report_contrib_register/__openerp__.py:
```diff
@@ -0,0 +1,36 @@
+
+{
+    'name': 'Payroll Contribution Register Report',
+    'version': '1.0',
```
Please adjust version number.

saltonmassally committed:
Cosmetic changes
adjustversion number
add try me on runbot button
add the bug tracker paragraph with predefined template.

saltonmassally commented:
@feketemihai - changes effected

feketemihai commented:
Could you change also the variables from readme with the correct ones. Thanks.

feketemihai commented on hr_payroll_report_contrib_register/__openerp__.py:
```diff
@@ -0,0 +1,37 @@
+    'author': "Salton Massally <smassally@idtlabs.sl>, "
+               "Odoo Community Association (OCA)",
+    'website': 'http://idtlabs.sl',
+    'license': 'AGPL-3',
```
Remove one license key line, is throwing a pylint error.

feketemihai commented:
@tarzan0820 @dufresnedavid Do somebody from you continue work on this, do you still need it for v8, there are too many modules on v8, which we don't have a continuation for a long time, so if nobody wants to continue them i think we should close it.

feketemihai commented:
Please reopen the PR if it's important to you.
